src/Lib/Canvas.vala

```diff
@@ -101,8 +100,32 @@ public class Akira.Lib.Canvas : Goo.Canvas {
         window.event_bus.set_focus_on_canvas.connect (on_set_focus_on_canvas);
         window.event_bus.request_escape.connect (on_set_focus_on_canvas);
         window.event_bus.insert_item.connect (on_insert_item);
+        // Make The canvas the drop target
+        Gtk.drag_dest_set (this, Gtk.DestDefaults.ALL, targets, Gdk.DragAction.COPY);
+        drag_data_received.connect(this.on_drag_data_received);
+    }
+    // Making The canvas recive the dropped file
+    private const Gtk.TargetEntry[] targets = {{"text/uri-list",0,0}};
+    private void on_drag_data_received (Gdk.DragContext drag_context, int x, int y, 
+        Gtk.SelectionData data, uint info, uint time){
+        // Loop through list of Files
+        foreach(string uri in data.get_uris ()){
+            string file = uri.replace("file://","").replace("file:/","");
+            file = Uri.unescape_string (file);
+            // Creating a Glib.File using a file path
+            var img_File = GLib.File.new_for_path(file);
+            var file_img_mangager = new Akira.Lib.Managers.ImageManager(img_File,GLib.Random.int_range(0,100));
+            // See if File is an Image File
+            if(Akira.Utils.Image.is_valid_image (img_File)){
+                // Tell Items Manager that we want to insert an Image
+                window.items_manager.insert_image(file_img_mangager);
+                // Insert Image to Canvas
+                window.items_manager.insert_item(x,y,file_img_mangager,null);
+            }
+        }
+        // Telling the Canvas to Stop the drag operation
+        Gtk.drag_finish (drag_context, true, false, time);
```

Comment:
The formatting of this method is incorrect.
Please use vala-lint to follow the proper code formatting we use in the whole app.
https://github.com/vala-lang/vala-lint